### ISSUE TITLE
Radio buttons not working with numeric value

### ISSUE THREAD

**Issue by PeterDraex** [fabian-hiller/modular-forms#273]
I've read the docs and implemented a form with radio buttons, but when you click on a radio button, the form state is not updated.

Here is a simplified code example.
```ts
const entities = [
  { id: 1, label: 'one' },
  { id: 2, label: 'two' },
];

type RadioForm = {
  entityID: number;
};
```

```tsx
<For each={entities}>
  {(entity) => (
    <Field name="entityID" type="number">
      {(field, props) => (
        <label>
          <input
            {...props}
            type="radio"
            value={entity.id}
            checked={field.value == entity.id}
          />
          {entity.label}
        </label>
      )}
    </Field>
  )}
</For>
Current value: {getValue(radioForm, 'entityID')}
```

Manually adding `onChange={(e) => setValue(radioForm, 'entityID', parseInt(e.target.value))}` to the input fixes the issue, but I'm guessing that's not the correct use, since no other input type requires manually setting the new value.

The [full example is available on Stackblitz](https://stackblitz.com/edit/modular-forms-solid-jfbhuqss?file=src%2Froutes%2Fradio.tsx).

Is this a bug?

**Comment by fabian-hiller** [comment]
What framework are you using? I am happy to share a working example from our [playground](https://modularforms.dev/solid/playground/special) with you.

**Comment by PeterDraex** [comment]
I'm using SolidJS. Right, in the StackBlitz example above, I used the  playground as the basis. 

The issue is that when I use `number` it breaks the radio buttons. I am assuming numeric values should be working too?

**Comment by fabian-hiller** [comment]
I don't think `<input type="radio />"` supports the `.valueAsNumber` property. When using `number`, the library doesn't do any transformation itself, it just reads the value of the HTML element.

**Comment by PeterDraex** [comment]
I see. Would a PR that implements conversions be welcomed? I think this is an unnecessary caveat

**Comment by fabian-hiller** [comment]
No, because I plan to rewrite the library. I will take this into account when rewriting. For now, I recommend to stick with `string` and transform it afterwards.

**Comment by fabian-hiller** [comment]
I reopen it so that I don't forget this when rewriting the library.

**Comment by brandonpittman** [comment]
Glad I found this. I got stuck on this for a while this morning.
